src/components/Booking/index.tsx

```diff
@@ -1,93 +1,151 @@
-const defaultTheme = createTheme();
+const Booking = () => {
+  const [shifts, setShifts] = useState<Shift[]>([]); 
+  const navigate = useNavigate(); 
 
-export default function Booking() {
-  // const user = React.useContext(UserContext);
+  useEffect(() => {
+    
+    const fetchShifts = async () => {
+      const token = localStorage.getItem("site");
+      if (!token) {
+        alert("Please login to view shifts.");
+        navigate("/login");
+        return;
+      }
+    
+      try {
+        console.log("Fetching shifts with token:", token);
+        const response = await fetch("/api/auth/shifts", {
+          method: "GET",
+          headers: {
+          //  "Content-Type": "application/json",
+            Authorization: token,
+          },
+        });
+    
+        // console.log("Response status:", response.status);
+        
+        if (!response.ok) {
+          const errorData = await response.json();
+          console.error("API Error:", response.status, errorData);
+  throw new Error(errorData.message || "API request failed");
+          // alert(errorData.message || "Unable to fetch shifts. Please try again.");
+          return;
+        }
+    
+        const data: Shift[] = await response.json();
+        setShifts(data);
+        console.log("Fetched shifts:", data);
+      } catch (error) {
+        console.error("Error fetching shifts:", error);
+        alert(`.`);
+        
+      }
+    };
 
-  // console.log(user ,"USER CONTEXT")
-  const auth = useAuth();
-  console.log("AUTh", auth.isAuthenticated)
-  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    const data = new FormData(event.currentTarget);
-    console.log({
-      email: data.get('email'),
-      password: data.get('password'),
-    });
-  };
-  const Item = styled(Paper)(({ theme }) => ({
-    backgroundColor: theme.palette.mode === 'dark' ? '#1A2027' : '#fff',
-    ...theme.typography.body2,
-    padding: theme.spacing(2),
-    textAlign: 'center',
-    color: theme.palette.text.secondary,
-  }));
-  const handleClick = () => {
-    console.info('You clicked the Chip.');
+    fetchShifts();
+  }, [navigate]);
+
+  const handleBookShift = async (shiftId: string) => {
+    console.log("ShiftId passed to handleBookShift:", shiftId);
+    const token = localStorage.getItem("site");
+    if (!token) {
+      alert("Please login to book a shift.");
+      navigate("/login");
+      return;
+    }
+    
+    try {
+      const response = await fetch(`/api/auth/book`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: token,
+        },
+        body: JSON.stringify({ ShiftId : shiftId }),
+      });
+
+      if (!response.ok) {
+        const errorData = await response.json();
+        console.error("Error booking shift try again and again:", errorData.message || "Unknown error");
```

Comment:
we can rather have wordings like ` There was an error confirming a shift, please try again at later time`